### DISCUSSION TITLE
FoxESS E/F Series F5000

### DISCUSSION THREAD

**Discussion by rjhazeld** [nathanmarlor/foxess_modbus#220] [General]
My installation has two inverters each serving a string of panels on opposite roofs. One is an H1-5.0-E which I have succeeded in setting up using a 410s after some help from canton7. The second inverter is an F5000 which has the same communications connector and RS485 connections. I have a second 410s set up identically to the one for the H1. I ran the modpoll command from the wiring guide with this result:
![image](https://user-images.githubusercontent.com/10947568/236483420-76c45ca4-587f-40bd-b64e-afc0b1d6e0cd.png)
This may be related to getting the inverter correctly setup, the CT menu item has options CT, Meter, and Disable. Currently only the 410s is connected via pins 3 & 4. Setting the CT option to 'meter' results in an error message 'meter fault'. Setting to CT clears the meter fault but doesn't change the modpoll result.
Canton7 indicated that a number of modpoll commands would need to be run and it may well be that the command I used wasn't quite right. I haven't attempted to add the F5000 to Home Assistant yet. What should I do next toward getting the F5000 data into Home Assistant? 
The relevant pages of the manual covering communications device installation and the function tree
![f5000 manual 20-21](https://user-images.githubusercontent.com/10947568/236488071-568e8c65-3fa5-45e8-82d3-3a713dcffe94.jpg)
![f5000 manual 14-15](https://user-images.githubusercontent.com/10947568/236488081-680e1bc5-200f-42f9-907e-9dd10027a534.jpg)
![f5000 manual 16-17](https://user-images.githubusercontent.com/10947568/236488085-7fb90f1a-4820-4234-96cd-97585ec563c4.jpg)

**Comment by canton7** [comment]
So, the CT is an analogue connection to your CT clamp, and the METER 485 is an RS485 connection to a smart meter, which the inverter can use instead of a CT clamp. Neither of those will expose a modbus interface.

From a quick flick through a manual online, I can't see any indication that the E- and F-series expose a modbus interface.

**Comment by rjhazeld** [comment]
The section at the top of page 16 suggests that a PC can acquire data with RS485. The section starts RS485/Meter followed by sections one for RS 485 and then one Meter(optional). Implying that meter and RS485 are separate facilities off the same pins. Having said that one of the diagrams showing wiring to a meter is annotated meter 485.

**Reply by canton7** [reply]
Hmm, true. Try reading holding registers (with `-t 4`), eg 30000

**Reply by canton7** [reply]
Although the invalid frame error looks suspicious. Maybe sniff the packet your adapter sends back, see what it contains. Maybe it uses the ASCII protocol instead of rtu or something

**Reply by canton7** [reply]
I think there's a modpoll parameter to change the frame format as well, might be worth playing with

**Comment by rjhazeld** [comment]
I'm afraid I am somewhat clueless where modpoll etc. is concerned ran these which may mean something to you

![image](https://user-images.githubusercontent.com/10947568/236522323-f905ccce-506b-4008-a349-cff99dc9a032.png)

I have to do a video call to my 96 year old father now will do some more digging later on.

**Reply by canton7** [reply]
You cut out the commands you ran there I'm afraid. It's address 30000 though, an extra 0.

I can suggest things to try but I'm afraid this will be largely you trying everything until something works

**Reply by rjhazeld** [reply]
Ok, I have found the manual for modpoll will have to wade through to understand what the various parameters do and try and find some other connection examples for other inverters.
Whilst there are responses the status page for the adapter has the conn status as listen/listen so I guess the initial connection fails and may be why there is no data output. I'll let you know how I get on.

**Comment by rjhazeld** [comment]
Haven't got results similar to those from the H1-5.0-E yet. I have downloaded a trial version of modbusinspector that has a raw page that is seeing output, I think, from the F5000. 
![modbusinspector raw](https://github.com/nathanmarlor/foxess_modbus/assets/10947568/49066940-c14e-4cd4-a166-0ebeb7379ece)
 It looks as though the 410s is connecting then the connection times out perhaps waiting for a log in string. I will be away for a few days but wonder whether the lines in the screen shot are interesting.

**Reply by canton7** [reply]
There's no concept of "logging in" with modbus.

That looks like a series of ModbusTCP frames, but it's trying to do (in sequence):

1. A read of 100 input coils from address 0, which the inverters don't support
2. A read of 100 discrete inputs from address 0, which the inverters don't support
3. A read of 100 holding registers from address 10000 (but the addresses in that range are input registers)
4. A read of 100 input registers from address 30000 (but the addresses in that range are holding registers)

The invalidness of the reads aside, the problem is that you're not getting a response at all.

**Comment by rand12345** [comment]
Have you tried sniffing the RS485 line rather than polling?

I have a feeling these inverters do not use modbus, they use a protocol which is used to exchange data between the inverter and cloud services. You can easily confirm this by monitoring the RS485 line and checking for a long byte array beginning 0x7e 0x7e ... and ending 0xe7 0xe7, these byte arrays may only occur once every few minutes.

If you find that this is the protocol you can read up on it [here.](https://community.home-assistant.io/t/foxess-inverter-modbus/459313/79)

**Comment by rjhazeld** [comment]
To date I haven't provoked a useful response from the F5000. As you will have gathered the technicalities are beyond my normal experience. I have just obtained a usb - rs485 interface and tried to get a response via quickmodbus but I guess that should enable me to look for traffic too. There are so many variables it is extremely difficult without some guidance from FOXess. In that regard I have been trying to get that help. I finally convinced someone from the UK support team that it was unreasonable and possibly a breach of some regulations to print the diagrams of inverter, line with RS485 and PC and a single sentence that says data can be read in the manual and not provide the basic information required to access the data.. He said he was going to refer to their software engineer for more information and would get back to me...we'll see.  I am not sure that there is RS485 output switched on by default as it is with the H1. Also pins 3 & 4 are annotated both RS485 and meter on the F5000 where they are solely RS485 on the H1.  I have asked for the necessary information to enable one to read the data e.g, protocol, addresses, registers. 

I'll see if I can find anything with the protocol you mention too.

**Comment by rjhazeld** [comment]
Apologies for the lack of communication since June. I have been in a rather unsatisafctory exchange of emails with Foxess UK Support to determine whether or not the F5000 can in fact be connected by RS485 and the data read. As Canton7 thought it does not, despite the manual indicating that it does.  This came from Foxess Product Management and Customer Solutions in Germany.  I have been trying to get a definitive response in respect of E series inverters (my daughter and son-in-law have one) without success but since the menu structure appears to be the same as the F series it seems unlikely.
I have taken the decision to add two further 'slices' of Icube battery to my system and will be haiving the F5000 replaced with an H1-5.0-E (without any batteries connected) which should mean I can get cosistennt data from both sets of panels via this integration.
